apps/prairielearn/src/lib/question-render.ts

```diff
@@ -559,29 +594,28 @@ export async function getAndRenderVariant(
 
   if (locals.instance_question) {
     await manualGrading.populateRubricData(locals);
-    await async.eachSeries(locals.submissions, manualGrading.populateManualGradingData);
+    await async.eachSeries(submissions, manualGrading.populateManualGradingData);
   }
 
   if (locals.question.type !== 'Freeform') {
     const questionJson = JSON.stringify({
-      questionFilePath: locals.calculationQuestionFileUrl,
-      questionGeneratedFilePath: locals.calculationQuestionGeneratedFileUrl,
-      effectiveQuestionType: locals.effectiveQuestionType,
-      course: locals.course,
-      courseInstance: locals.course_instance,
+      questionFilePath: urls.calculationQuestionFileUrl,
+      questionGeneratedFilePath: urls.calculationQuestionGeneratedFileUrl,
+      effectiveQuestionType,
+      course,
+      courseInstance: course_instance,
       variant: {
-        id: locals.variant.id,
-        params: locals.variant.params,
+        id: variant.id,
+        params: variant.params,
       },
-      submittedAnswer:
-        locals.showSubmissions && locals.submission ? locals.submission.submitted_answer : null,
-      feedback: locals.showFeedback && locals.submission ? locals.submission.feedback : null,
-      trueAnswer: locals.showTrueAnswer ? locals.variant.true_answer : null,
-      submissions: locals.showSubmissions ? locals.submissions : null,
+      submittedAnswer: submission?.submitted_answer ?? null,
+      feedback: submission?.feedback ?? null,
+      trueAnswer: resultLocals.showTrueAnswer ? variant.true_answer : null,
+      submissions,
```

Comment:
Just double-checking that this is ok (since I'm not that familiar with v2 questions), but if there are no submissions this changes from null to an empty array. Is this ok?

Reply:
It's probably fine, but best not to mess with v2 😓 I restored the original behavior in f9c00a6.